doc/developer-guides/hld/hv-partitionmode.rst

```diff
@@ -0,0 +1,358 @@
+.. _partition-mode-hld:
+
+Partition mode
+##############
+
+ACRN is type-1 hypervisor that supports running multiple guest operating
+systems (OS). Typically, the platform BIOS/boot-loader boots ACRN, and
+ACRN loads single or multiple guest OSes. Refer to :ref:`hv-startup` for
+details on the start-up flow of the ACRN hypervisor.
+
+ACRN supports two modes of operation: Sharing mode and Partition mode.
+This document describes ACRN's high-level design for Partition mode
+support.
+
+.. contents::
+   :depth: 2
+   :local:
+
+Introduction
+************
+
+ACRN, in partition mode, is used to provide guests with exclusive access
+to cores, memory, cache, and IO devices. There is no resource sharing
+between the guests. ACRN expects static partitioning, by developers, of
+resources either by code modification for guest configuration or through
+compile-time config options. All the devices exposed to the guests are
+either physical resources or emulated in the hypervisor. So, there is no
+need for device-model and Service OS. :numref:`pmode2vms` shows a
+partition mode example of two VMs with exclusive access to physical
+resources.
+
+.. figure:: images/partition-image3.png
+   :align: center
+   :name: pmode2vms
+
+   Partition Mode example with two VMs
+
+Guest info
+**********
+
+ACRN uses multi-boot info passed from the platform boot-loader to know
+the location of each guest kernel in memory. ACRN creates a copy of each
+guest kernel into each of the guests' memory. Current implementation of
+ACRN requires developers to specify kernel parameters for the guests as
+part of guest configuration. ACRN picks up kernel parameters from guest
+configuration and copies them to the corresponding guest memory.
+
+.. figure:: images/partition-image18.png
+   :align: center
+
+ACRN set-up for guests
+**********************
+
+Cores
+=====
+
+ACRN requires the developer to specify the number of guests and the
+cores dedicated for each guest. Also the developer needs to specify
+the physical core used as the Boot Strap Processor (BSP) for each guest. As
+the processors are brought to life in the hypervisor, it checks if they are
+configured as BSP for any of the guests. If a processor is BSP of any of
+the guests, ACRN proceeds to build the memory mapping for the guest,
+mptable, E820 entries, and zero page for the guest. As described in
+`Guest info`_, ACRN creates copies of guest kernel and kernel
+parameters into guest memory. :numref:`partBSPsetup` explains these
+events in chronological order.
+
```

Comment:
Can we also describe the operating mode from with the BSP starts with? I remember this is different in partition mode and sharing mode.

Reply:
@junjiemao1 I think that difference is captured. In sharing mode, only the platform BSP proceeds with launching SOS. In partition mode, APs also check if they are any of the guests BSP.